src/ORM/Search/SearchContext.php

```diff
@@ -411,7 +423,7 @@ public function removeFilterByName($name)
      */
     public function getFields()
     {
-        return $this->fields;
+        return $this->fields ?? FieldList::create();
```

Comment:
I'm not changing `addField` - that's not in scope.

Reply:
> If you call getFields 5 times in a row, you'll get 5 different instances. Is that what's expected?

I was going to say yes but then I realised that because `FieldList` is mutable it would give unexpected results. I'll change the way I'm dealing with the problem described in https://github.com/silverstripe/silverstripe-framework/pull/11049#discussion_r1398654282 and https://github.com/silverstripe/silverstripe-framework/pull/11049#discussion_r1398654764